platform/android/SCsub

```diff
@@ -82,7 +98,12 @@ if lib_arch_dir != "":
     env_android.CommandNoCache(out_dir + "/libgodot_android.so", lib, Move("$TARGET", "$SOURCE"))
 
     stl_lib_path = (
-        str(env["ANDROID_NDK_ROOT"]) + "/sources/cxx-stl/llvm-libc++/libs/" + lib_arch_dir + "/libc++_shared.so"
+        str(env["ANDROID_NDK_ROOT"])
+        + "/toolchains/llvm/prebuilt/"
+        + host_subpath
+        + "/sysroot/usr/lib/"
+        + triple_target_dir
+        + "/libc++_shared.so"
```

Comment:
Could be simplified a bit with f-strings maybe (excluding the first one as it's a bit of a complex expression to embed in the string IMO).
```suggestion
        str(env["ANDROID_NDK_ROOT"])
        + f"/toolchains/llvm/prebuilt/{host_subpath}/sysroot/usr/lib/{triple_target_dir}/libc++_shared.so"
```

Or if we want to include everything on a long line, this should work:
```python
stl_lib_path = f"{env['ANDROID_NDK_ROOT']}/toolchains/llvm/prebuilt/{host_subpath}/sysroot/usr/lib/{triple_target_dir}/libc++_shared.so"
```